pandas/core/ops.py

```diff
@@ -1706,7 +1708,8 @@ def f(self, other, axis=default_axis, level=None, fill_value=None):
             if fill_value is not None:
                 self = self.fillna(fill_value)
 
-            return self._combine_const(other, na_op, try_cast=True)
+            pass_op = op if lib.is_scalar(other) else na_op
```

Comment:
you are checking for a scalar here and above?

Reply:
It's kind of annoying.  If `lib.is_scalar(other)` then we will be dispatching to the `Series` op, in which case we want to pass the "raw" op (e.g.` operator.add`) and not the wrapped op `na_op`.

This PR handles only scalars since that is a relatively easy case.  A few PRs down the road we'll have all these ops dispatch to series, at which point this won't be necessary.